File: happy.js
function isHappy(num) {
  let slow = num;
  let fast = num;

  do {
    slow = nextNumber(slow);
    fast = nextNumber(nextNumber(fast));
  } while (slow != fast);

  return slow == 1;
}

function nextNumber(num) {
  let sum = 0;
  while (num > 0) {
    let digit = num % 10;
    sum += digit * digit;
    num = Math.floor(num / 10);
  }
  return sum;
}

console.log(isHappy(19));

File: happy.c
#include <stdio.h>

int next_number(int num) {
  int sum = 0;
  while (num > 0) {
    int digit = num % 10;
    sum += digit * digit;
    num /= 10;
  }
  return sum;
}

int is_happy(int num) {
  int slow = num;
  int fast = num;
  do {
    slow = next_number(slow);
    fast = next_number(next_number(fast));
  } while (slow != fast);
  return slow == 1;
}

int main() {
  int num;
  printf("Enter a number: ");
  scanf("%d", &num);
  if (is_happy(num)) {
    printf("%d is a happy number.\n", num);
  } else {
    printf("%d is not a happy number
